types/index.d.ts

```diff
@@ -4,19 +4,88 @@ import http = require('http');
 import https = require('https');
 import stream = require('stream');
 
+/**
+ * Supported compression types.
+ */
+export enum CompressionType {
+  GZIP = 'gzip',
+  DEFLATE = 'deflate'
+}
+
 export interface IRequestOptions {
+
+  /**
+   * Parameters for query string. ({ param: 1 } produces ?param=1 etc.) 
+   */
   qs?: { [key: string]: string | number | boolean | Array<string | number> };
+  
+  /**
+   * An object that consumes the logging requests
+   */
+  logger: any;
```

Comment:
Looks like this can be made more strict as according to https://github.com/laurisvan/request-promise-lite/blob/785f39b3b951c8c347ee9cab6433f51b3c397842/src/ConsoleLogger.js#L10

@laurisvan anything else preventing this PR from being merged?

Reply:
Ok, this should be now addressed in the latest commit. Looking forward to get this committed @laurisvan - let me know if there's anything that needs to be polished change.

And thanks @dashuser33 for the feedback!